smolyay/grid.py

```diff
@@ -0,0 +1,362 @@
+
+import abc
+import numpy
+from smolyay.basis import (BasisFunction, ChebyshevFirstKind,
+                           BasisFunctionSet, NestedBasisFunctionSet)
+import itertools
+
+
+class IndexGridGenerator(abc.ABC):
+    """Grid points for making a surrogate.
+
+    Depending on the approach and dimensionality
+    , a set of grid points, and their corresponding
+    basis functions are generated. These grid points
+    and functions can then be then used for approximation
+    of complex systems.
+    """
+
+    def __init__(self, basis_set):
+        self._basis_set = basis_set
```

Comment:
If you do it this way, it will trigger the setter (in case there is anything special you want the setter to do)
```suggestion
        self.basis_set = basis_set
```